app/controllers/api/flavors_controller.rb

```diff
@@ -1,4 +1,28 @@
 module Api
   class FlavorsController < BaseController
+    def create_resource(_type, _id, data)
+      manager_id = parse_id(data['ems'], :providers)
+      raise 'Must specify a valid manager_resource href or id' unless manager_id
```

Comment:
this is looking good! 😄  one minor thing - the error message does not match the data it is expecting. Perhaps change it to `data['manager_resource']`?